### PR TITLE
PHP 8.* compatibility

### PR DESCRIPTION
**Pull request by lukasbesch**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
     if: "!contains(github.event.head_commit.message, '[ci skip]')"
     strategy:
       matrix:
-        php: ['7.3', '7.4']
+        php: ['7.3', '7.4', '8.0', '8.1', '8.2']
```
Comment (codepuncher) [inline .github/workflows/ci.yml]: Thanks for the PR @lukasbesch !

Could you remove the old 7.x versions?
Best we don't support that any more

File: composer.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
     }
   ],
   "require": {
-    "php": "^7.3",
+    "php": "^7.3|^8.0",
```
Comment (codepuncher) [inline composer.json]: may as well just
`composer require php:^8.0`; 7.x is dead

**Review by codepuncher — APPROVED** [review]
Thanks for the PR @lukasbesch!

**Comment by codepuncher** [comment]
Anything else you want to add to this before I merge @lukasbesch?

**Comment by lukasbesch** [comment]
We could add type declarations instead of doc blocks for the class properties.

Will look into #37 (or #38 now) again at some point, because I just saw it being flagged in my IDE.

Other than that, I am just picking up an older project with Sage 9 at the moment, and will see if I run into more (actual) issues on PHP 8 and the current WordPress version.

So maybe leave it unmerged for now, I will give you an update in the following days.

**Comment by codepuncher** [comment]
> We could add type declarations instead of doc blocks for the class properties.

I fully agree with this.

> So maybe leave it unmerged for now

Up to you mate; I'm happy to merge now if you want to submit a new PR for types.

**Review by codepuncher — APPROVED** [review]

**Pull request merged**